src/datatablerow/utils.jl

```diff
@@ -0,0 +1,191 @@
+# Rows grouping.
+# Maps row contents to the indices of all the equal rows.
+# Used by groupby(), join(), nonunique()
+immutable RowGroupDict{T<:AbstractDataTable}
+    "source data table"
+    dt::T
+    "number of groups"
+    ngroups::Int
+    "row hashes"
+    rhashes::Vector{UInt}
+    "hashindex -> index of group-representative row"
+    gslots::Vector{Int}
+    "group index for each row"
+    groups::Vector{Int}
+    "permutation of row indices that sorts them by groups"
+    rperm::Vector{Int}
+    "starts of ranges in rperm for each group"
+    starts::Vector{Int}
+    "stops of ranges in rperm for each group"
+    stops::Vector{Int}
+end
+
+# "kernel" functions for hashrows()
+# adjust row hashes by the hashes of column elements
+function hashrows_col!(h::Vector{UInt}, v::AbstractVector)
+    @inbounds for i in eachindex(h)
+        h[i] = hash(v[i], h[i])
+    end
+    h
+end
+
+function hashrows_col!{T<:Nullable}(h::Vector{UInt}, v::AbstractVector{T})
+    @inbounds for i in eachindex(h)
+        h[i] = isnull(v[i]) ?
+               hash(Base.nullablehash_seed, h[i]) :
```

Comment:
I didn't want to imply that that function was called, just that it was consistent with it. I don't think changing this should generate any failures since the hash functions are completely under our control. Have you made the same change to all places where `hash` is involved in the PR?